packages/commonwealth/client/scripts/state/api/topics/createTopic.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import app from 'state';
+import Topic from 'models/Topic';
+import { useMutation } from '@tanstack/react-query';
+import { ApiEndpoints, queryClient } from 'state/api/config';
+
+interface CreateTopicProps {
+  name: string;
+  description: string;
+  telegram?: string;
+  featuredInSidebar: boolean;
+  featuredInNewPost: boolean;
+  tokenThreshold: string;
+  defaultOffchainTemplate: string;
+}
+
+const createTopic = async ({
+  name,
+  description,
+  telegram,
+  featuredInSidebar,
+  featuredInNewPost,
+  tokenThreshold,
+  defaultOffchainTemplate,
+}: CreateTopicProps) => {
+  const response = await axios.post(`${app.serverUrl()}/createTopic`, {
+    name,
+    description,
+    telegram,
+    featured_in_sidebar: featuredInSidebar,
+    featured_in_new_post: featuredInNewPost,
+    token_threshold: tokenThreshold || '0',
+    default_offchain_template: defaultOffchainTemplate,
+    jwt: app.user.jwt,
+    chain: app.activeChainId(),
+  });
+
+  return new Topic(response.data.result);
+};
```

Comment:
We should always have API call as a separate function as this gives us a nice readability. This function always in private